frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -778,6 +807,7 @@ type dispatchState struct {
 	buildArgs      []instructions.KeyValuePairOptional
 	commands       []command
 	ctxPaths       map[string]struct{}
+	namedCtxState  *namedContextState
```

Comment:
I think we can avoid the `namedContextState` type. I would define `includePaths map[string]struct{}` in here and explicitly call `SetLocalOpts` like you had in the previous version to make it clear what is the order of gathering paths and marshaling. If you need a helper function then that is fine but I think `normalizeContextPaths` already does the job.

Reply:
Yea this makes sense.